### PR TITLE
Pizza Overhaul: Reworks, resprites, and refactors pizza entirely

### PR DESCRIPTION
**Pull request by MylieDaniels**
[CATERING] [GAME OBJECTS] [REWORK] [CODE QUALITY]
### Pizza Overhaul
Pizza has been completely redone, and now has no hardcoded recipes, only a dynamic system for making custom pizzas with any sauce and toppings you can get your oven mitts on.

#### New visuals!
All pizza sprites have been updated.
image
image

#### Sauces and cheese!
Rather than being limited to tomato juice, pizzas now support up to 50 units of any reagent as a sauce. They will recolor their sauce to match the mixture, and render cheese over this sauce. Cheese can be applied either as a wedge for a standard cheese pizza, a wheel for deep dish, or ignored entirely. You don't need to add sauce, either, but I encourage you to. Please don't make none pizzas, I haven't added left beef to the game yet.
Here we have a pizza with royal initrobeedril sauce, in various states of progress:
image

#### New topping customization!
The toppings on a pizza now visually display where they were placed, shrunken by 50% and randomly rotated. They can be removed at any time from the dough, but once the pizza is cooked, they are deleted and their image, effects, and reagents are all that remains - see below for the details of how that is handled.
Here I've added an apple, a lettuce, and some old gloves I found to a royal initrobeedril pizza:
image

#### Refactored code!
Pizza objectives are still a thing for chefs, but now calculate on the pizza itself, rather than tying to the oven. They were also given some safety checks, to ensure they don't get caught in infinite loops at roundstart- much the same as cake objectives have. Pizzas have been repathed, and split into children where logical.

#### The technical bits
Pizza now holds up to 10 w_class of items, each of which must be no larger than SMALL, or 2 w_class. If a topping is a food item, the final pizza will gain an equal amount of healing and filling. If a topping is not a food item, it adds half it's weight class in filling, and removes a bit of healing.
Each topping can transfer at most 100 units of reagents to the final pizza, the sauce can transfer 50, and 20 units of bread will be present. A deep dish pizza adds 40 units (one cheese wheel) of cheese, while a regular adds 5 units. The final pizza will also have 40 units of "stuffed crust space", which allows for poisonings that aren't as visible as the color-advertising sauce or the toppings which state what they are.
When the pizza is baked, all food effects are pooled, and the four with the most occurrences pass down. However, the pizza adds 2 "false occurrences" of Gross Burps to this pool, along with 2 of Max Hp Up (20) if cheesed, or 2 of Max Hp Up (40) if deep dish. This means that making a pizza with the exact buffs you want requires overcoming the Gross Burps effect and likely using cheese.
When the pizza is cut, the healing, filling, and reagents are split amongst the slices.
The only way to eat a whole pizza is Matter Eater, which adds 20 units of salmonella if it was uncooked.
The pizza dough can be cooked in an oven for 9 seconds on HIGH, as before, or exposed to a temperature above 3200 Celsius.
Using a pizza dough inhand will now spin it, with a forty percent chance of failure if you aren't a chef. Ten or more successful spins in a row makes the dough 10% more healthy!

## Why's this needed? 
Pizza was stale, both codewise, mechanically, and visually. This is another step towards a more dynamic, less oven-recipe-based cooking system.

## Changelog 

```changelog
(u)MylieDaniels
(+)Pizza has been completely reworked. To get started, roll some dough. To finish it up, bake for 9 seconds on HIGH, or find a way to expose to a terribly high heat. The middle? That's entirely up to you.
```

### CODE REVIEW

**Comment by Sbmhawk** [comment]
Most people eat the crust last, is it possible for the stuffed crust reagent to only apply on thr last bite or two?

Thought it would be interesting like that

**Comment by Glamurio** [comment]
Ninja Turtles rejoice

**Comment by MylieDaniels** [comment]
> Most people eat the crust last, is it possible for the stuffed crust reagent to only apply on thr last bite or two?
> 
> Thought it would be interesting like that

Not without making a separate reagent pool, which I'd rather avoid doing due to how many things would need to be special cased for it.

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drink/pizza.dm [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+	Eat(mob/M as mob, mob/user, by_matter_eater = TRUE)
+		if(!by_matter_eater)
+			return 0
+		var/obj/item/reagent_containers/food/snacks/pizza/unbaked_pizza = src.bake_pizza(TRUE)
+		return unbaked_pizza.Eat(M, user, TRUE)
+
+	temperature_expose(datum/gas_mixture/air, temperature, volume)
+		if (temperature >= T0C+3200) // syndicate zippos and raging plasmafires, for laughs
+			var/obj/item/reagent_containers/food/snacks/pizza/baked_pizza = src.bake_pizza()
+			baked_pizza.quality = 5
```
Comment (Tarmunora) [inline code/modules/food_and_drink/pizza.dm:156]: could use some an audio/chatbox cue when this happens

File: code/mob/living/critter/bee.dm [hunk 2]
```diff
@@ -262,12 +262,10 @@
 		var/obj/item/reagent_containers/food/snacks/pizza/floor_pizza = locate() in honeyTurf
 		var/obj/item/reagent_containers/food/snacks/ingredient/honey/honey
```
Comment (ZeWaka) [inline code/mob/living/critter/bee.dm:263]: please fix this pathing being wrong while your'e here, it should end at snacks

**Review by ZeWaka — COMMENTED** [review]
reviewed everything except pizza.dm

File: code/modules/food_and_drink/pizza.dm [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+			src.cheese = SafeGetOverlayImage("cheese", 'icons/obj/foodNdrink/food_ingredient.dmi', "pizzacheese[src.cheesy]")
+		UpdateOverlays(src.sauce, "sauce", TRUE)
+		UpdateOverlays(src.cheese, "cheese", TRUE)
+		..()
+
+	attackby(obj/item/W, mob/user, params)
+		if (!src.reagents.total_volume && !length(src.contents) && !src.cheesy && (iscuttingtool(W) || issawingtool(W) || issnippingtool(W))) // make tortillas if untouched
+			boutput(user, SPAN_NOTICE("You cut [src] into smaller pieces."))
+			for(var/i = 1, i <= 3, i++)
+				new /obj/item/reagent_containers/food/snacks/ingredient/tortilla(get_turf(src))
```
[the remark this answers is not in the document]
Comment (MylieDaniels) [reply code/modules/food_and_drink/pizza.dm:58]: It is but this is also the old behavior and only way to get tortillas aough

File: code/modules/food_and_drink/pizza.dm (hunk 1 above)
Comment (MylieDaniels) [reply code/modules/food_and_drink/pizza.dm:156]: Sure thing!

File: code/modules/food_and_drink/pizza.dm [hunk 4]
```diff
@@ -0,0 +1,593 @@
...
+				if (istype(food))
+					for (var/effect in food.food_effects)
+						if (temp_food_effects[effect])
+							temp_food_effects[effect]++
+						else
+							temp_food_effects[effect] = 1
+					baked_pizza.heal_amt += food.heal_amt / 12
+					baked_pizza.fill_amt += food.fill_amt
+				else
+					baked_pizza.heal_amt -= topping.w_class / 18
```
[the remark this answers is not in the document]
Comment (MylieDaniels) [reply code/modules/food_and_drink/pizza.dm:256]: It is intentional, yes, to make "wrench pizza" heal for less. I'll comment it!

File: code/modules/food_and_drink/pizza.dm [hunk 5]
```diff
@@ -0,0 +1,593 @@
+/obj/item/reagent_containers/food/snacks/ingredient/pizza_base
+	name = "pizza dough"
+	desc = "A good start to a delicious pizza."
+	icon_state = "pizzabase"
+	initial_volume = 50
+	custom_food = FALSE
+	var/sauce_color = "#d24300"
+	var/cheesy = 0
+	var/image/sauce = null
+	var/image/cheese = null
```
[the remark this answers is not in the document]
Comment (MylieDaniels) [reply code/modules/food_and_drink/pizza.dm:10]: Sure thing, will do.

File: code/mob/living/critter/bee.dm (hunk 2 above)
Comment (MylieDaniels) [reply code/mob/living/critter/bee.dm:263]: Absolutely!

File: code/modules/food_and_drink/pizza.dm [hunk 6]
```diff
@@ -153,6 +153,8 @@
 	temperature_expose(datum/gas_mixture/air, temperature, volume)
 		if (temperature >= T0C+3200) // syndicate zippos and raging plasmafires, for laughs
 			var/obj/item/reagent_containers/food/snacks/pizza/baked_pizza = src.bake_pizza()
+			for(var/mob/M in AIviewers(baked_pizza))
+				boutput(M, SPAN_NOTICE("\The [baked_pizza.name] roasts from heat!"))
```
Comment (ZeWaka) [inline code/modules/food_and_drink/pizza.dm:157]: visible message

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drink/pizza.dm (hunk 6 above)
Comment (MylieDaniels) [reply code/modules/food_and_drink/pizza.dm]: Somehow I forgot about that existing...

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**